resources/views/userprofile.blade.php

```diff
@@ -52,101 +83,13 @@
     <br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><br><hr>
 
     <script>
-        function closeModal() {
-            document.getElementById('settingsModal').style.display = 'none';
-        }
-
-        function openModal() {
-            document.getElementById('settingsModal').style.display = 'block';
+        function closeModal(modalName) {
```

Comment:
bootstrap are suport pentru data attributes. data-toggle=modal, data-target=#id-of-your-target. nu trebuie sa faci tu functii de openModal / closemodal si sa folosesti onclick.

https://getbootstrap.com/docs/5.3/components/modal/#via-data-attributes